source/GameWindow.cpp

```diff
@@ -108,6 +151,12 @@ bool GameWindow::Init()
 	// Settings that must be declared before the window creation.
 	Uint32 flags = SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE | SDL_WINDOW_ALLOW_HIGHDPI;
 
+#ifdef __linux__
+	// Force fullscreen for the Steam Deck.
+	if(IsSteamDeck())
+		Preferences::Set("fullscreen", true);
+#endif
```

Comment:
Since we don't sync preferences across machines, I don't see any need for detecting any specific platforms if we make the default behavior to open fullscreen. And if we ever do sync preferences across machines, I certainly think we will want certain preferences to never sync across machines, and fullscreen is one of those that should be machine-specific.

And further, I disagree entirely that we should force fullscreen if the user has not set the fullscreen preference. (So what if the game runs better in fullscreen? The user said no, so don't run fullscreen.)

Reply:
> And further, I disagree entirely that we should force fullscreen if the user has not set the fullscreen preference.

That is kinda true and I agree. Although in this case it's because the windowed game has some blank spaces at the top and bottom for some reason (which we should fix properly at some later time).

Reply:
This adversely affects steam deck where there is no easily accessible shortcut to toggle fullscreen.  Anybody who has tried ES on (launched once) on their steam deck forever has this setting without the game intervening or the user (unlikely) having a USB-C keyboard.

There’s no scenario on steam deck where the game shouldn’t be fullscreen.  SteamOS 3 already handles fullscreen apps well and it’s intended they be fullscreen.

Reply:
> This adversely affects steam deck


This game is not designed for Steam Deck, a luxury platform that will remain far more performant than the minimum requirements for the game.

> Anybody who has tried ES (launched once) on their steam deck forever has this setting without the game intervening or the user (unlikely) having a USB-C keyboard.

I'm not in a place to look at it currently, but by adopting a class enum for the setting, rather than a plain boolean, we can handle the "you haven't played this version yet" scenario.

> There’s no scenario on steam deck where the game shouldn’t be fullscreen.  SteamOS 3 already handles fullscreen apps well and it’s intended they be fullscreen. 

Again, ES is not designed for Steam Deck. I'm against adding specific device platform code because it is a maintenance burden for us, e.g. "I don't have this esoteric hardware, so now I can't safely modify this code and verify what this code is doing."